### PR TITLE
Fix nested types output from createStrictAPI() being unintentionally marked as required

### PR DESCRIPTION
**Pull request by itsdouges**
This pull request updates types and adds some tests for this specific use case.

### CODE REVIEW

File: packages/react/src/create-strict-api/__tests__/generics.test.tsx [hunk 1]
```diff
@@ -5,6 +5,58 @@ import type { XCSSProp } from './__fixtures__/strict-api-recursive';
 import { css, cssMap } from './__fixtures__/strict-api-recursive';
 
 describe('createStrictAPI()', () => {
+  it('should mark all styles as optional in css()', () => {
+    const styles = css({
+      '&:hover': {},
```
Comment (itsdouges) [inline packages/react/src/create-strict-api/__tests__/generics.test.tsx:10]: This was a type violation previously.

File: packages/react/src/create-strict-api/index.ts [hunk 2]
```diff
@@ -6,11 +6,11 @@ type PseudosDeclarations = {
...
 };
 
-type EnforceSchema<TObject> = {
-  [P in keyof TObject]?: P extends keyof CompiledSchema
-    ? TObject[P] extends Record<string, unknown>
-      ? EnforceSchema<TObject[P]>
-      : TObject[P]
+type EnforceSchema<TSchema> = {
+  [P in keyof TSchema]?: P extends keyof CompiledSchema
+    ? TSchema[P] extends Record<string, any>
```
Comment (itsdouges) [inline packages/react/src/create-strict-api/index.ts:11]: Main fix: `unknown` -> `any`

Comment (kylorhall-atlassian) [reply packages/react/src/create-strict-api/index.ts:11]: Interesting, do you know why this works? I've seen similar things, just never seen like a TS issue or explanation.

File: packages/babel-plugin/src/css-map/index.ts [hunk 3]
```diff
@@ -98,15 +98,15 @@ export const visitCssMapPath = (
         const { sheets, classNames } = transformCssItems(css, meta);
         totalSheets.push(...sheets);
 
-        if (classNames.length !== 1) {
+        if (classNames.length > 1) {
```
Comment (itsdouges) [inline packages/babel-plugin/src/css-map/index.ts:101]: This allows us to make empty styles now (at least for testing).

File: packages/babel-plugin/src/css-map/index.ts [hunk 4]
```diff
@@ -98,15 +98,15 @@ export const visitCssMapPath = (
...
           throw buildCodeFrameError(
             createErrorMessage(ErrorMessages.STATIC_VARIANT_OBJECT),
             property,
             meta.parentPath
           );
         }
 
-        return t.objectProperty(property.key, classNames[0]);
+        return t.objectProperty(property.key, classNames[0] || t.stringLiteral(''));
```
Comment (kylorhall-atlassian) [inline packages/babel-plugin/src/css-map/index.ts:109]: Got it, so having `'&:hover'` empty threw this?

**Review by kylorhall-atlassian — APPROVED** [review]
Seems straightforward from my understanding, and I know there's tests for `'&:hover': { … }` having valid css in them, so LGTM.

File: packages/babel-plugin/src/css-map/index.ts (hunk 4 above)
Comment (itsdouges) [reply packages/babel-plugin/src/css-map/index.ts:109]: Yeah exactly.

Comment (itsdouges) [reply packages/babel-plugin/src/css-map/index.ts:109]: > Interesting, do you know why this works? I've seen similar things, just never seen like a TS issue or explanation.

I think it's an edge case with complex nested types (interface + omits + ?). At a glace `unknown` should be fine but in this case it's obviously not.

I'm not sure!

File: packages/react/src/create-strict-api/index.ts (hunk 2 above)
Comment (itsdouges) [reply packages/react/src/create-strict-api/index.ts:11]: I think it's an edge case with complex nested types (interface + omits + ?). At a glace unknown should be fine but in this case it's obviously not.

I'm not sure!

**Pull request merged**